app/javascript/components/item-view.vue

```diff
@@ -0,0 +1,100 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import { useNotification } from '@kyvg/vue3-notification';
+import purchasesApi from '../api/purchases';
+import type { Item } from '../api/items';
+import type { Review } from '../api/reviews';
+import ReviewListCard from './review-list-card.vue';
+import ReviewSummaryCard from './review-summary-card.vue';
+
+type Props = {
+  item: Item,
+  reviews: Review[]
+}
+
+const props = defineProps<Props>();
+
+const loading = ref(false);
+const { notify } = useNotification();
+
+async function buy() {
+  loading.value = true;
+  try {
+    await purchasesApi.create(props.item.id);
+    notify({ text: 'Genial, recibimos tu orden! Te contactaremos para coordinar', type: 'success' });
+  } catch (error) {
+    notify({ text: 'Ups, ocurrió un error! Inténtalo de nuevo', type: 'error' });
+  } finally {
+    loading.value = false;
+  }
+}
+</script>
+
+<template>
+  <div>
+    <a
+      class="inline-flex items-center gap-2 rounded-full bg-zinc-100 px-3 py-2 text-sm"
+      href="/"
+    >
+      <inline-svg
+        class="h-4 w-4 fill-white"
+        :src="require('images/icon/back-arrow.svg')"
+      />
+      Volver
+    </a>
+    <div class="mt-12 md:flex md:gap-4">
+      <div>
+        <img :src="item.image.sm.url">
+      </div>
+      <div>
+        <h1 class="text-lg font-bold text-zinc-800">
+          {{ item.name }}
+        </h1>
+        <p class="text-3xl text-zinc-800">
+          {{ item.price }}
+        </p>
+
+        <div class="md:mt-4">
+          <p class="text-base">
+            {{ item.description }}
+          </p>
+          <base-button
+            class="mt-4 w-full rounded-full"
+            :disabled="loading"
+            @click="buy"
+          >
+            Comprar
+          </base-button>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div
+    v-if="reviews.length > 0"
+    class="space-y-6"
+  >
+    <div>
+      <h2 class="text-xl font-bold text-zinc-800">
+        Opiniones del producto
+      </h2>
+      <review-summary-card
+        :count="item.reviewsCount"
+        :average="item.reviewsAverage"
```

Comment:
🚫 **[tsc]** <2322> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Type 'number' is not assignable to type 'string'.